ledger/participant-integration-api/src/test/lib/scala/platform/indexer/ha/IndexerStabilitySpec.scala

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.indexer.ha
+
+import com.daml.ledger.api.health.HealthStatus
+import com.daml.ledger.api.testing.utils.AkkaBeforeAndAfterAll
+import com.daml.ledger.resources.ResourceContext
+import com.daml.logging.LoggingContext
+import com.daml.platform.store.DbType
+import com.daml.platform.store.backend.StorageBackend
+import org.scalatest.concurrent.Eventually
+import org.scalatest.flatspec.AsyncFlatSpec
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.time.{Millis, Seconds, Span}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait IndexerStabilitySpec
+    extends AsyncFlatSpec
+    with Matchers
+    with AkkaBeforeAndAfterAll
+    with Eventually {
+
+  // To be overriden by the spec implementation
+  def jdbcUrl: String
+
+  // The default EC is coming from AsyncTestSuite and is serial, do not use it
+  implicit val ec: ExecutionContext = system.dispatcher
+  private implicit val loggingContext: LoggingContext = LoggingContext.ForTesting
+
+  behavior of "redundant parallel indexers"
+
+  it should "correctly work in high availability mode" in {
+    val updatesPerSecond = 10 // Number of updates per second produced by the read service
+    val indexerCount = 8 // Number of concurrently running indexers
+    val restartIterations = 8 // Number of times the indexer should restart
+
+    implicit val rc: ResourceContext = ResourceContext(ec)
+
+    info(s"Creating indexers fixture with $indexerCount indexers")
+    IndexerStabilityTestFixture
+      .owner(
+        updatesPerSecond,
+        indexerCount,
+        jdbcUrl,
+        materializer,
+      )
+      .use[Unit] { indexers =>
+        var abortedIndexer: Option[ReadServiceAndIndexer] = None
+        (1 to restartIterations).foreach(_ => {
+          // Assert that there is exactly one indexer running
+          val activeIndexer = eventuallyAfterRecovery { indexers.runningIndexer }
+          info(s"Indexer ${activeIndexer.readService.name} is running")
+
+          // The indexer should appear "healthy"
+          eventually {
+            assert(
+              activeIndexer.indexing.currentHealth() == HealthStatus.healthy,
```

Comment:
I am not sure about this check here (also not sure I understood the intention correctly)
An HA protected indexer will be healthy as waiting for the indexer lock.